packages/components/src/molecules/Rating/Rating.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { forwardRef, useState } from 'react'
+import type { ReactNode } from 'react'
+import type { HTMLAttributes } from 'react'
+
+import List from '../../atoms/List'
+import { Icon, IconButton } from '../..'
+
+export interface RatingProps extends HTMLAttributes<HTMLUListElement> {
+  /**
+   * The current value of the rating, based on the quantity of child elements.
+   */
+  value: number
+  /**
+   * A React component that will be rendered as an icon.
+   */
+  icon: ReactNode
+  /**
+   * Specifies that this input should be controled by the user.
+   */
+  actionable?: boolean
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+}
+
+export interface RatingItemProps {
+  'data-fs-rating-item'?: 'full' | 'partial' | 'empty'
+  'data-testid'?: string
+}
+
+const Rating = forwardRef<HTMLUListElement, RatingProps>(
+  function Rating(
+    { children, testId = 'fs-rating', value, actionable, icon, ...otherProps },
+    ref
+  ) {
+    const [rating, setRating] = useState(value - 1)
+    const [hover, setHover] = useState(0)
+
+    return (
+      <List
+        ref={ref}
+        data-fs-rating
+        data-fs-rating-actionable={actionable}
+        data-testid={testId}
+        {...otherProps}
+      >
+        {Array.from({ length: 5 }).map((_, index: number) => {
+
+          const fillCheck = (itemValue: number) => {
+            if (index <= (hover || rating)) {
```

Comment:
The initial value is the value (current rate) - 1, so when you'll check if should fill full, you can add the 1, but this is broken when is partial, I'll be left the details in another comment.
```suggestion
            if (index <= (hover || rating + 1)) {
```